src/prefect/utilities/google.py

```diff
@@ -0,0 +1,62 @@
+"""
+Utility functions for interacting with Google Cloud.
+"""
+from google.cloud import bigquery
+from google.cloud import storage
+from google.oauth2.service_account import Credentials
+
+
+def get_google_client(submodule, credentials: dict = None, project: str = None):
+    """
+    Utility function for loading Google Client objects from a given set of credentials.
+
+    Args:
+        - submodule: a Python submodule with a Client attribute
+        - credentials (dict, optional): a dictionary of Google credentials used to initialize the Client; if
+            not provided, will attempt to load the Client using ambient environment settings
+        - project (str, optional): the Google project to point the Client to; if not provided, Client defaults
+            will be used
+
+    Returns:
+        - Client: an initialized and authenticated Google Client
+    """
+    Client = getattr(submodule, "Client")
```

Comment:
This abstraction is usually true but I'm not confident it's always true. PubSub has two different clients for example (one for pub, one for sub).

Reply:
Yea that's very fair; my vote would be to cross that bridge when we get to it if that works for you